bpf/process/generic_calls.h

```diff
@@ -1088,6 +1088,46 @@ FUNC_INLINE int filter_args(struct msg_generic_kprobe *e, int selidx, bool is_en
 	return 0;
 }
 
+_Static_assert(5 == MAX_SELECTORS, "update selidx_next()");
+FUNC_INLINE int next_selidx(struct msg_generic_kprobe *e, int selidx)
+{
+	int idx = selidx + 1;
+
+	switch (idx) {
+	case 0:
+		if (e->sel.active[idx & MAX_SELECTORS_MASK])
+			break;
+		idx++;
+		fallthrough;
+	case 1:
+		if (e->sel.active[idx & MAX_SELECTORS_MASK])
+			break;
+		idx++;
+		fallthrough;
+	case 2:
+		if (e->sel.active[idx & MAX_SELECTORS_MASK])
+			break;
+		idx++;
+		fallthrough;
+	case 3:
+		if (e->sel.active[idx & MAX_SELECTORS_MASK])
+			break;
+		idx++;
+		fallthrough;
+	case 4:
+		if (e->sel.active[idx & MAX_SELECTORS_MASK])
+			break;
+		idx++;
+		fallthrough;
+	case 5:
+		if (e->sel.active[idx & MAX_SELECTORS_MASK])
+			break;
+		idx++;
+	}
```

Comment:
your switch case looks like a loop no?
```suggestion
	for (int i = idx; i <= 5; i++) {
		if (e->sel.active[idx & MAX_SELECTORS_MASK])
			break;
		idx++;
	}
```

Reply:
Indeed, but I wasn't sure how well the compiler would unroll it given the (int i = idx), so I just unrolled it on my own.